pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -851,6 +851,13 @@ func (p *DNSProxy) ServeDNS(w dns.ResponseWriter, request *dns.Msg) {
 	}
 
 	targetServerID := identity.ReservedIdentityWorld
+	if option.Config.IsDualStack() {
+		if targetServerIP.To4() == nil {
```

Comment:
If `To4()` returns nil, then we set the ID to world-ipv4? Shouldn't the condition be

```suggestion
		if targetServerIP.To4() != nil {
```

[1]

Reply:
Good catch.